crates/paimon/src/spec/types.rs

```diff
@@ -157,7 +163,40 @@ impl FromStr for DataType {
     type Err = Error;
 
     fn from_str(_: &str) -> Result<Self, Self::Err> {
-        todo!()
+        todo!();
+    }
+}
+
+impl Serialize for DataType {
```

Comment:
Hi, this has native support in serde. Please add `#[serde(untagged)]` for `DataType` instead.

Reference: https://serde.rs/enum-representations.html#untagged